### PR TITLE
Devectorize parameters in evaluator

### PR DESCRIPTION
**Pull request by polvalente**

### CODE REVIEW

File: nx/test/nx/defn_test.exs [hunk 1]
```diff
@@ -2115,6 +2115,50 @@ defmodule Nx.DefnTest do
...
+               |> String.replace(~r/\s/, "")
+
+      assert Nx.Defn.jit_apply(
+               fn x, y ->
+                 [x, y] = Nx.broadcast_vectors([x, y])
+                 {x, y}
+               end,
+               [x, y],
+               compiler: Evaluator
+             ) == 1
```
Comment (polvalente) [inline nx/test/nx/defn_test.exs:2159]: @josevalim I ran into this issue while working on the `cond` PR. In short, it seems that the concrete parameter value that's being fetched in the evaluator to be fed to the backend is returning with vectorized axes.

File: nx/lib/nx/defn/evaluator.ex [hunk 2]
```diff
@@ -305,7 +305,7 @@ defmodule Nx.Defn.Evaluator do
               "cannot pass a tensor expression as argument to defn, got: #{inspect(tensor)}"
 
       %Nx.Tensor{} = tensor ->
-        {tensor, caches}
+        {Nx.devectorize(tensor), caches}
```
Comment (polvalente) [inline nx/lib/nx/defn/evaluator.ex:308]: This fixes the test, but I don't think it's the correct solution, right?

Comment (josevalim) [reply nx/lib/nx/defn/evaluator.ex:308]: Ideally we wouldn't have anything vectorized here. If it is vectorized, it means it is leaking somewhere. But this is helpful. I will take a look!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**